lib/archethic/transaction_chain/transaction.ex

```diff
@@ -953,4 +949,61 @@ defmodule Archethic.TransactionChain.Transaction do
         |> Enum.map(&CrossValidationStamp.cast/1)
     }
   end
+
+  defp get_movements_from_token_transaction(tx_address, tx_content) do
+    case Jason.decode(tx_content) do
+      {:ok, json} ->
+        cond do
+          ExJsonSchema.Validator.valid?(@token_creation_schema, json) ->
+            get_movements_from_token_creation(tx_address, json)
+
+          ExJsonSchema.Validator.valid?(@token_resupply_schema, json) ->
+            get_movements_from_token_resupply(json)
+
+          true ->
+            []
+        end
+
+      {:error, _} ->
+        []
+    end
+  end
+
+  defp get_movements_from_token_creation(tx_address, json) do
+    recipients = json["recipients"] || []
+    fungible? = json["type"] == "fungible"
+
+    Enum.map(recipients, fn r = %{"to" => address_hex, "amount" => amount} ->
+      token_id = r["token_id"] || if fungible?, do: 0, else: 1
```

Comment:
Here the token id should be always present if the token is non_fungible otherwise the user could send multiple time the same NFT. (It will result in unsufficient funds error)